summingbird-core/src/main/scala/com/twitter/summingbird/planner/Node.scala

```diff
@@ -78,12 +78,42 @@ case class SourceNode[P <: Platform[P]](override val members: List[Producer[P, _
   override def shortName(sanitize: String => String) = NodeIdentifier("Source" + collapseNamedNodes(sanitize))
 }
 
-case class Dag[P <: Platform[P]](originalTail: TailProducer[P, _], producerToPriorityNames: Map[Producer[P, Any], List[String]], tail: TailProducer[P, _], producerToNode: Map[Producer[P, _], Node[P]],
-    nodes: List[Node[P]],
-    nodeToName: Map[Node[P], String] = Map[Node[P], String](),
-    nameToNode: Map[String, Node[P]] = Map[String, Node[P]](),
-    dependenciesOfM: Map[Node[P], List[Node[P]]] = Map[Node[P], List[Node[P]]](),
-    dependantsOfM: Map[Node[P], List[Node[P]]] = Map[Node[P], List[Node[P]]]()) {
+/**
+ * Summingbird converts a Directed Acyclic Graph(DAG) of Producers into a DAG of Nodes.
```

Comment:
It's probably worth calling out that Dag objects aren't (typically) constructed directly, but rather through the companion object helpers, which also generate (long) names for each Node.